diff --git a/node_modules/http-cache-semantics/index.js b/node_modules/http-cache-semantics/index.js
index 56c2602..26cd423 100644
--- a/node_modules/http-cache-semantics/index.js
+++ b/node_modules/http-cache-semantics/index.js
@@ -395,6 +395,13 @@ module.exports = class CachePolicy {
     evaluateRequest(req) {
         this._assertRequestHasHeaders(req);
 
+        // Do not let max-stale override entries zeroed for security reasons.
+        if (!this.storable() || this._rescc['no-cache'] ||
+            (this._isShared && this._resHeaders['set-cookie'] && !this._rescc.public && !this._rescc.immutable) ||
+            this._resHeaders.vary === '*' || (this._isShared && this._rescc['proxy-revalidate'])) {
+            return this._evaluateRequestMissResult(req);
+        }
+
         // In all circumstances, a cache MUST NOT ignore the must-revalidate directive
         if (this._rescc['must-revalidate']) {
             return this._evaluateRequestMissResult(req);
